tests/db_functions/tests.py

```diff
@@ -547,3 +547,74 @@ def test_now(self):
             ['How to Time Travel'],
             lambda a: a.title
         )
+
+    def test_length_transform(self):
+        try:
+            CharField.register_transform(Length, 'length')
```

Comment:
We'll need a corresponding `register_transform()` context manager also. I'm wondering if we should pull out the existing context manager and put it beside the RegisterLookupMixin as a useful utility. At the moment it's hidden within a specific test module. I'd prefer not to reimplement (copy/paste) in this file just to save a 3 or so lines, but I'd be happy to import it.